vscode/vscode/vscode.mli

```diff
@@ -674,78 +582,77 @@ module ConfigurationTarget : sig
     | Workspace
     | WorkspaceFolder
 
-  val t_of_js : Ojs.t -> t
-
-  val t_to_js : t -> Ojs.t
+  include Js.T with type t := t
 end
 
 module WorkspaceConfiguration : sig
-  type t
-
-  type configurationTarget =
-    [ `ConfigurationTarget of ConfigurationTarget.t
-    | `Bool of bool
-    ]
+  include Js.T
 
-  type inspectResult =
+  type 'a inspectResult =
     { key : string
-    ; defaultValue : Jsonoo.t option
-    ; globalValue : Jsonoo.t option
-    ; workspaceValue : Jsonoo.t option
-    ; workspaceFolderValue : Jsonoo.t option
-    ; defaultLanguageValue : Jsonoo.t option
-    ; globalLanguageValue : Jsonoo.t option
-    ; workspaceLanguageValue : Jsonoo.t option
-    ; workspaceFolderLanguageValue : Jsonoo.t option
-    ; languageIds : string option
+    ; defaultValue : 'a option
+    ; globalValue : 'a option
+    ; workspaceValue : 'a option
+    ; workspaceFolderValue : 'a option
+    ; defaultLanguageValue : 'a option
+    ; globalLanguageValue : 'a option
+    ; workspaceLanguageValue : 'a option
+    ; workspaceFolderLanguageValue : 'a option
+    ; languageIds : string list option
     }
 
-  val get : t -> section:string -> unit -> Jsonoo.t option
+  val get : (module Js.T with type t = 'a) -> t -> section:string -> 'a option
+
+  val get_json : t -> section:string -> Jsonoo.t option
+
+  val get_default :
+       (module Js.T with type t = 'a)
```

Comment:
It would be nice to introduce an alias like this:

```
  type 'a js = (module T with type t = 'a)
```

To avoid having to type the module all the time.

By the way, did you consider just using a record instead:
```ocaml
type 'a js =
  { from : Ojs.t -> 'a
  ; to_ : 'a -> Ojs.t
  }
```

I don't think it's all that important, but records are generally much more convenient to work with.

Reply:
I added the alias.

I considered doing it with records, but I think it's easier with first class modules because all of the modules in `vscode.ml` already conform to `Js.T` module type. It's as simple as just packing the module into a value. Making it a record requires extra code without much benefit in my opinion.

Using first class modules also adds some symmetry between generic types (applying a module to a functor) and generic functions (applying a first class module to a function).

Reply:
Yeah, that sounds good. First class modules become more of a pain when you need lightweight combinators like `val iso : 'a js -> ('a -> 'b) -> ('b -> 'a) -> 'b js.t`, but we have no use for them yet.